MHV-68471: Made sure Need Help component is always present on Meds List page

vmenshutin-bylight opened the pull request:
## Summary

- Rx: Move NeedHelp component to the bottom of the tree to make sure it is always present

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-68471

## Testing done

- Manual testing

## What areas of the site does it impact?

/my-health/medications

## Acceptance criteria

AC1: Need Help component on the Meds list needs to be present if there are no meds in the list, for all different errors, if the users don't have access, etc.. IT SHOULD ALWAYS BE PRESENT ON THE MEDS LIST NO MATTER WHAT

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

vmenshutin-bylight committed:
MHV-68471: Add "Need Help" component on every Meds list page

robertbylight commented on src/applications/mhv-medications/containers/Prescriptions.jsx:
```diff
@@ -874,6 +873,7 @@ const Prescriptions = () => {
             )}
           </>
         )}
+        {removeLandingPage && !isLoading && <NeedHelp page={pageType.LIST} />}
```
looks good, question about when to show the NeedHelp component. If the need help section needs to always be present, wouldn't we want it to also be present when the spinner is displayed? or is that inconsequential?

vmenshutin-bylight replied:
<img width="645" alt="image" src="https://github.com/user-attachments/assets/a56c413b-10ed-4b80-b138-8154b39eba6b" />

Commit:
e2e need help section for empty rx and api failure

added e2e for validating need help section for no prescriptions and api call failure
